addons/pos_self_order/static/src/app/pages/kiosk_product_list_page/product_list_page.js

```diff
@@ -0,0 +1,321 @@
+                    container.scrollLeft = selectedLeft;
+                } else if (selectedRight > containerRight) {
+                    container.scrollLeft = selectedRight - container.clientWidth;
+                }
+            }
+
+            this.initCategoriesDragToScroll(this.categoryListRef.el, this.categoryListRef);
+            this.initCategoriesDragToScroll(
+                this.subCategoryContainerRef.el,
+                this.subCategoryListRef
+            );
+        });
+
+        onWillUnmount(() => {
+            if (this.scrollMouseListeners) {
+                for (const { onMouseMove, onMouseUp } of this.scrollMouseListeners) {
+                    window.removeEventListener("mousemove", onMouseMove);
+                    window.removeEventListener("mouseup", onMouseUp);
+                }
+            }
+        });
+    }
+
+    back() {
+        this.dialog.add(CancelPopup, {
+            title: _t("Cancel order"),
+            confirm: () => {
+                this.router.navigate("categories");
+            },
+        });
+    }
+
+    selectCategory(category) {
+        this.selfOrder.currentCategory = category;
+        if (!category.parent_id) {
+            this.toggleSubCategoryPanel();
+        }
+        this.productListRef.el?.scrollTo({ top: 0 });
+    }
+
+    isProductAvailable(product) {
+        if (product.pos_categ_ids.length === 0) {
+            return true;
+        }
+        return product.pos_categ_ids.some((categ) => this.selfOrder.isCategoryAvailable(categ.id));
+    }
+
+    get topSelectedCategory() {
+        return this.selectedCategory?.parent_id || this.selectedCategory;
+    }
+
+    get selectedCategory() {
+        return this.selfOrder.currentCategory;
+    }
+
+    getSubCategories() {
+        const currentCategory = this.selfOrder.currentCategory;
+        if (!currentCategory) {
+            return [];
+        }
+        if (currentCategory.parent_id) {
+            return currentCategory.parent_id.child_ids;
+        }
+        return currentCategory.child_ids || [];
+    }
+
+    toggleSubCategoryPanel() {
+        const el = this.subCategoryContainerRef.el;
+        const nextSubCategories = this.getSubCategories();
+        if (this.state.subCategories.length > 0 && nextSubCategories.length === 0) {
+            el.classList.remove("show");
+            const oldSelectedCat = this.selectedCategory;
+            const self = this;
+            el.addEventListener("transitionend", function handler(e) {
+                if (oldSelectedCat === self.selectedCategory) {
+                    self.state.subCategories = [];
+                }
+                el.removeEventListener("transitionend", handler);
+            });
+            return;
+        } else if (nextSubCategories.length === 0 && this.state.subCategories.length === 0) {
+            return;
+        }
+
+        this.state.subCategories = nextSubCategories;
+        el.classList.add("show");
+    }
```

Comment:
It's a bit weird to do this in JS, isn't there a way to do it directly in the XML via selectedCategory.child_ids? 
The record is reactive, it will display the panel automatically.